Utils/Dataflow/019_esFormat/esFormat.php

```diff
@@ -41,18 +93,50 @@ function convertIndexToLowerCase(&$a) {
 }
 
 function getAction($row) {
-  global $DEFAULT_ACTION;
+  /* Determine action for the document.
+
+  Action is 'update' if {'_update': true} key-value pair is 
+  present in the document. Otherwise, default value is returned.
+
+  :param row: document for which the action should be determined
+  :type row: array
+
+  :return: action
+  :rtype: str
+  */
+  global $ACTION;
 
   if (isset($row['_update']) and $row['_update'] === true) {
     $action = 'update';
   } else {
-    $action = $DEFAULT_ACTION;
+    $action = $ACTION;
```

Comment:
Somehow this part makes me a bit nervous.

`$ACTION` *is* defined somewhere in the "main" program code, but it is not obvious that this function will only be used after it happens.

What do you think?

Maybe add some kind of "safety wire" here: if `$ACTION` is not set, fall back to the `$DEFAULT_ACTION`?
I know it is exactly what happens [here](https://github.com/PanDAWMS/dkb/pull/292/files#diff-c5f3f98d4013266cbee7633d5d2f7ec8R252):
```
if (!(isset($ACTION))) $ACTION = $DEFAULT_ACTION;
```
..., and that situation with `$ACTION` is just the same as with `$ES_INDEX`, used in `constructActionJson($row)`; but still...

Reply:
As you correctly point out, this situation - "we use global variable in a function without checking whether it was correctly initialized" - is present in other places of this script. Furthermore, it also exists in [at least one other place](https://github.com/PanDAWMS/dkb/blob/019-minor-fixes/Utils/Dataflow/071_esConsistency/consistency.py#L107) in this repository (by the way, there seems to be an error in documentation in that place, will look into it). In light of such situation, I'm not sure that adding a safety measure in this single place is a way to go.
Will make a Trello card about the situation in general, or maybe address the situation's instances in this stage in this PR, since it fits into "fix various minor problems" of the description.

Reply:
> Will make a Trello card about the situation in general

https://trello.com/c/ZUZvlz3n

> and maybe address the situation's instances in this stage in this PR

Decided not to, since this looks like a general issue that warrants some thinking.

Reply:
OK then, I'll mark this one as resolved.